frontend/src/concepts/trustyai/utils.ts

```diff
@@ -15,3 +23,60 @@ export const formatListResponse = (x: BaseMetricListResponse): BiasMetricConfig[
     thresholdDelta: m.request.thresholdDelta,
     unprivilegedAttribute: m.request.unprivilegedAttribute.value,
   }));
+
+export const getTrustyStatusState = (
+  crFetchState: FetchState<TrustyAIKind | null>,
+  successDetails?: UseTrustyBrowserStorage,
+): TrustyStatusStates => {
+  const [cr, loaded, error] = crFetchState;
+
+  if (error) {
+    return { type: TrustyInstallState.ERROR, message: error.message };
+  }
+
+  if (!loaded) {
+    return { type: TrustyInstallState.LOADING_INITIAL_STATE };
+  }
+
+  if (!cr) {
+    // No CR, uninstalled
+    return { type: TrustyInstallState.UNINSTALLED };
+  }
+
+  if (cr.metadata.deletionTimestamp) {
+    return { type: TrustyInstallState.UNINSTALLING };
+  }
+
+  // Have a CR, getting specific state
+  const availableCondition = getConditionForType(cr, 'Available');
+  if (availableCondition?.status === 'True' && cr.status?.phase === 'Ready') {
+    // Installed and good to go
+    return {
+      type: TrustyInstallState.INSTALLED,
+      showSuccess: !!successDetails?.showSuccess,
+      onDismissSuccess: successDetails?.onDismissSuccess,
+    };
+  }
+
+  const dbAvailableCondition = getConditionForType(cr, 'DBAvailable');
+  if (dbAvailableCondition?.status === 'False') {
+    // Some sort of DB error -- try to show specifically what it is
+    return {
+      type: TrustyInstallState.ERROR,
+      message: `${dbAvailableCondition.reason ?? 'Unknown reason'}: ${
+        dbAvailableCondition.message ?? 'Unknown error'
+      }`,
+    };
+  }
```

Comment:
How many conditions can there be. can you just go over all of them dynamically (excluding available) and set status based on if any of those are false?

Reply:
I don't want to treat them blindly like that... there are some that speak to InferenceServices being available -- and that's not a critical failure like that. Ready can be true and that condition can be false.

Reply:
Also worth mentioning, this PR is in support of that specific condition -- which is why I called it out specifically.